src/staticdata_utils.c

```diff
@@ -907,6 +907,12 @@ static jl_array_t *jl_verify_edges(jl_array_t *targets, size_t minworld)
             jl_array_ptr_1d_push(_jl_debug_method_invalidation, loctag);
             loctag = jl_box_int32((int32_t)i);
             jl_array_ptr_1d_push(_jl_debug_method_invalidation, loctag);
+            if (matches == jl_nothing) {
+                assert(sig);
+                int ambig = 0;
+                matches = jl_matching_methods((jl_tupletype_t*)sig, jl_nothing,
+                                              -1, 0, minworld, &min_valid, &max_valid, &ambig);
+            }
```

Comment:
This only gets triggered from `matches == jl_nothing` which comes from https://github.com/JuliaLang/julia/blob/7ba7e326293bd3eddede81567bbe98078e81f775/src/staticdata_utils.c#L872-L876

It basically  restores the version that was used before the efficiency-improvement in #48841. If this isn't correct, let's put the correct thing here, but it seemed to pass muster before the efficiency-improvement.

Reply:
The change in https://github.com/JuliaLang/julia/pull/48841 alters the algorithm selection from the `-1` algorithm to the positive limit algorithm. What you probably want is to make the `jl_array_len(expected)` parameter selectable based on how many conflicts you want returned. E.g. up to `INT32_MAX` conflicts, or just `+4` or such until it gives us and returns "too many"

Reply:
OK, I added just one extra space. I think we only need to identify one cause; if there turn out to be multiples, folks can re-run the analysis after fixing the one that got identified. I pushed this change.

**However**, then I tried modifying the test to add a second invalidating method. Then I got `nothing` in that slot again 😢. Big bummer.

I haven't delved into how `jl_matching_methods` works in a while, but I am surprised it returned `nothing` rather than a partial list. Is there anything "safe" short of re-running the analysis with -1? While I'm OK with only *returning* a single source of invalidation, returning `nothing` is simply unacceptable. As you point out above, we might have to re-run the setdiff. But I think I'd rather do that than miss causes of invalidation. What are your thoughts?

Reply:
We can either do `+1` to get the list if there was only 1 cause, or `+N` to get up to N causes (before declaring "a lot") or INT32_MAX to get all causes.

Reply:
OK, and `INT32_MAX` would be acceptable? Or does that cause a performance hit?

Reply:
I went ahead and just changed it in the original computation. But if instead you meant that I should use `INT32_MAX` to recompute the matches when logging, just LMK.